src/app/icd/server/ICDManager.cpp

```diff
@@ -439,26 +440,28 @@ void ICDManager::UpdateOperationState(OperationalState state)
             std::bind(&ICDManager::ShouldCheckInMsgsBeSentAtActiveModeFunction, this, std::placeholders::_1, std::placeholders::_2);
 #endif // CHIP_CONFIG_ENABLE_ICD_CIP
 
-        // When the active mode interval is 0, we stay in idleMode until a notification brings the icd into active mode
-        // unless the device would need to send Check-In messages
-        if (ICDConfigurationData::GetInstance().GetActiveModeDuration() > kZero
+        // When the ActiveModeDuration is set to 0, the ICDManager does not need to periodically transition to active mode.
+        // Instead, It can stay in idle mode until a notification, Report or other network event automatically toggles the ICD into
+        // active mode. The following conditions will schedule a transition to Active Mode after the Idle Mode duration expires.
+        // - An ActiveModeDuration interval must be respected.
+        // - The device state indicates to shorten its idle duration and report faster to provide better responsiveness
+        // - Check-In messages must be sent
+        if (configData.GetActiveModeDuration() > kZero || configData.ShouldUseShortIdle()
```

Comment:
I see why the config is public now, but I am wondering if the Config should be the one holding that logic.

Couldn't we have the manager make the decisions based on it's configs, rather than the config also holding the decision logic?